src/PRBProxy.sol

```diff
@@ -93,7 +89,8 @@ contract PRBProxy is
     /// @inheritdoc IPRBProxy
     function execute(address target, bytes calldata data) external payable override returns (bytes memory response) {
         // Check that the caller is either the owner or an envoy with permission.
-        if (owner != msg.sender && !permissions[msg.sender][target]) {
+        bool permission = registry.getPermissionByOwner({ owner: owner, envoy: msg.sender, target: target });
```

Comment:
You can save gas here; check `msg.sender` is the owner before the call to the registry to check the permission of `msg.sender` as an envoy. It's not necessary to do this call for the owner as they should always have permission, unless you want to make it possible for an owner to block themselves from using certain target contracts.

Reply:
Good catch!

We don't want to let the owner block themselves from using any targets, so we will make the call to `getPermissionByOwner` only after making the `owner` check.